File: app/src/main/java/org/wikipedia/dataclient/mwapi/CreateAccountResponse.java
package org.wikipedia.dataclient.mwapi;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class CreateAccountResponse extends MwResponse {
    @SuppressWarnings("unused") @Nullable private Result createaccount;

    @Nullable public String status() {
        return createaccount.status();
    }

    @Nullable public String user() {
        return createaccount.user();
    }

    @Nullable public String message() {
        return createaccount.message();
    }

    public boolean hasResult() {
        return createaccount != null;
    }

    public static class Result {
        @SuppressWarnings("unused,NullableProblems") @NonNull private String status;
        @SuppressWarnings("unused") @Nullable private String message;
        @SuppressWarnings("unused") @Nullable private String username;

        @NonNull public String status() {
            return status;
        }

        @Nullable public String user() {
            return username;
        }

        @Nullable public String message() {
            return message;
        }
    }
}

File: app/src/main/java/org/wikipedia/dataclient/mwapi/CreateAccountResponse.kt
package org.wikipedia.dataclient.mwapi

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class CreateAccountResponse : MwResponse() {

    private val createaccount: Result? = null

    val status get() = createaccount?.status.orEmpty()

    val user get() = createaccount?.username.orEmpty()

    val message get() = createaccount?.message.orEmpty()

    val messageCode get() = createaccount?.messageCode.orEmpty()

    @Serializable
    class Result {

        val status: String = ""
        val message: String = ""
        @SerialName("messagecode") val messageCode: String = ""
        val username: String = ""
    }
}
